File: bootstrap-ui-tests/src/test/java/uitest/pages/LoginPage.java
package uitest.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import uitest.commands.SeleniumCommands;


public class LoginPage {

    private final WebDriver driver;
    SeleniumCommands sc;

    @FindBy(css = "#login")
    private WebElement loginField;

    @FindBy(css = "#password")
    private WebElement passwordField;

    @FindBy(css = "button[type=submit]")
    private WebElement loginButton;

    public LoginPage(WebDriver driver) {
        this.driver = driver;
        sc = new SeleniumCommands(driver);
    }

    public void login(String login, String password) throws Exception {
        loginField.sendKeys(login);
        passwordField.sendKeys(password);
        loginButton.click();
        sc.waitForFinishLoading();
    }

    public void openLoginPage() throws Exception {
        driver.get("http://localhost:8080/#/login");
        sc.waitForFinishLoading();
    }
}

File: bootzooka-ui-tests/src/test/scala/uitest/pages/LoginPage.scala
package uitest.pages

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy
import uitest.commands.SeleniumCommands

class LoginPage(driver: WebDriver) {
  private val sc: SeleniumCommands = new SeleniumCommands(driver)
  val url = sc.URL + "login"

  @FindBy(name = "login")
  val loginField: WebElement = null

  @FindBy(name = "password")
  val passwordField: WebElement = null

  @FindBy(css = "button[type=submit]")
  val loginButton: WebElement = null

  def login(login: String, password: String) {
    loginField.sendKeys(login)
    passwordField.sendKeys(password)
    loginButton.click()
    sc.waitForFinishLoading()
  }

  def openLoginPage() {
    driver.get(url)
    sc.waitForFinishLoading()
  }
}